packages/core/test/icon/iconTests.tsx

```diff
@@ -14,20 +14,30 @@ import { Classes, Icon, IIconProps, Intent } from "../../src/index";
 
 describe("<Icon>", () => {
     it("iconSize=16 renders standard size", () =>
-        assertIconSize(<Icon iconName="graph" iconSize={Icon.SIZE_STANDARD} />, Icon.SIZE_STANDARD));
+        assertIconSize(<Icon icon="graph" iconSize={Icon.SIZE_STANDARD} />, Icon.SIZE_STANDARD));
 
     it("iconSize=20 renders large size", () =>
-        assertIconSize(<Icon iconName="graph" iconSize={Icon.SIZE_LARGE} />, Icon.SIZE_LARGE));
+        assertIconSize(<Icon icon="graph" iconSize={Icon.SIZE_LARGE} />, Icon.SIZE_LARGE));
 
     it("renders intent class", () =>
-        assert.isTrue(shallow(<Icon iconName="add" intent={Intent.DANGER} />).hasClass(Classes.INTENT_DANGER)));
+        assert.isTrue(shallow(<Icon icon="add" intent={Intent.DANGER} />).hasClass(Classes.INTENT_DANGER)));
 
-    it("renders iconName class", () => assertIcon(<Icon iconName="calendar" />, "calendar"));
+    it("renders icon name", () => assertIcon(<Icon icon="calendar" />, "calendar"));
 
-    it("supports prefixed iconName", () => assertIcon(<Icon iconName={IconClasses.AIRPLANE} />, "airplane"));
+    it("supports prefixed icon name", () => assertIcon(<Icon icon={IconClasses.AIRPLANE} />, "airplane"));
 
-    it("iconName=undefined renders nothing", () => {
-        const icon = shallow(<Icon iconName={undefined} />);
+    it("passes through icon element unchanged", () => {
```

Comment:
also leave that documentation here, to explain why we're supporting this weird prop